tests/cpp/dlr_data_transform_test.cc

```diff
@@ -147,6 +147,42 @@ TEST(DLR, DataTransformMultipleColumn) {
   }
 }
 
+TEST(DLR, DataTransformDateTime) {
+  dlr::DataTransform transform;
+  nlohmann::json metadata = R"(
+    {
+      "DataTransform": {
+        "Input": {
+          "ColumnTransform": [
+            {
+              "Type": "DateTime"
+            }
+          ]
+        }
+      }
+    })"_json;
+  EXPECT_TRUE(transform.HasInputTransform(metadata));
+
+  const char* data = R"([["Jan 3rd, 2018, 1:34am"], ["Feb 11th, 2012, 11:34:59pm"]])";
```

Comment:
Could you add some bad entries here to test that they are handled gracefully? Maybe like some empty input, just a number, a random word.